python/cuml/test/test_lars.py

```diff
@@ -0,0 +1,191 @@
+    "column_info",
+    [
+        unit_param([20, 10]),
+        quality_param([100, 50]),
+        stress_param([1000, 500])
+    ],
+)
+@pytest.mark.parametrize("normalize", [True, False])
+@pytest.mark.parametrize("precompute", [True, False, 'precompute'])
+def test_lars_model(datatype, nrows, column_info, precompute, normalize):
+    ncols, n_info = column_info
+    X_train, X_test, y_train, y_test = make_regression_dataset(
+        datatype, nrows, ncols, n_info
+    )
+
+    if precompute == 'precompute' or not normalize:
+        # Apply normalization manually, because the solver expects normalized
+        # input data
+        X_train, y_train, x_mean, x_scale, y_mean = \
+            normalize_data(X_train, y_train)
+        y_test = y_test - y_mean
+        X_test = (X_test - x_mean) / x_scale
+
+    if precompute == 'precompute':
+        precompute = np.dot(X_train.T, X_train)
+
+    params = {'precompute': precompute, 'normalize': normalize}
+
+    # Initialization of cuML's LARS
+    culars = cuLars(**params)
+
+    # fit and predict cuml LARS
+    culars.fit(X_train, y_train)
+
+    cu_score_train = culars.score(X_train, y_train)
+    cu_score_test = culars.score(X_test, y_test)
+
+    if nrows < 500000:
+        # sklearn model initialization, fit and predict
+        sklars = skLars(**params)
+        sklars.fit(X_train, y_train)
+
+        assert cu_score_train >= sklars.score(X_train, y_train) - 0.05
+        assert cu_score_test >= sklars.score(X_test, y_test) - 0.1
+    else:
+        assert cu_score_test > 0.95
+
+
+@pytest.mark.parametrize("datatype", [np.float32, np.float64])
+@pytest.mark.parametrize(
+    "nrows", [unit_param(500), quality_param(5000), stress_param(500000)]
+)
+@pytest.mark.parametrize(
+    "column_info",
+    [
+        unit_param([20, 10]),
+        quality_param([100, 50]),
+        stress_param([1000, 500])
+    ],
+)
+@pytest.mark.parametrize("precompute", [True, False])
+def test_lars_collinear(datatype, nrows, column_info, precompute):
+    ncols, n_info = column_info
+
+    X_train, X_test, y_train, y_test = make_regression_dataset(
+        datatype, nrows, ncols, n_info
+    )
+    n_duplicate = min(ncols, 100)
+    X_train = np.concatenate((X_train, X_train[:, :n_duplicate]), axis=1)
+    X_test = np.concatenate((X_test, X_test[:, :n_duplicate]), axis=1)
+
+    params = {"precompute": precompute, "n_nonzero_coefs": ncols + n_duplicate}
+    culars = cuLars(**params)
+    culars.fit(X_train, y_train)
+
+    assert culars.score(X_train, y_train) > 0.85
+    assert culars.score(X_test, y_test) > 0.85
+
+
+@pytest.mark.parametrize("datatype", [np.float32, np.float64])
+@pytest.mark.parametrize("params", [{"precompute": True},
+                                    {"precompute": False},
+                                    {"n_nonzero_coefs": 5},
+                                    {"n_nonzero_coefs": 2}])
+def test_lars_attributes(datatype, params):
+    X, y = load_boston(return_X_y=True)
```

Comment:
Maybe try one variant with `fit_intercept=False` (some of that codepath seems to be uncovered)

Reply:
Yes, I was already working on this yesterday, just got sidelined with other issues. Added the test.

After this only the memory errors handling is not covered. I will open a separate issue for that to be tasted later.